### ISSUE TITLE
Windows bind mount fails when used by tar in linux container

### ISSUE THREAD

**Issue by brucelowekamp** [docker/for-win#5649]
My Linux container in Docker desktop for windows is working for building etc.  But when tar is run on a large tree (make distcheck from ola open source package), it fails, first producing "file changed as we read it" error for all of the files, then eventually the volume mount becomes permanently disconnected "transport endpoint is not connected" error produced by the build and by the OS (i.e. "ls" doesn't work on the volume mount)

![image](https://user-images.githubusercontent.com/14317301/73462159-54a51500-4349-11ea-8e39-edade2f4e7cb.png)

Restarting Docker makes the volume visible again, but now trying to delete files produces 

![image](https://user-images.githubusercontent.com/14317301/73462356-afd70780-4349-11ea-8af7-46ee899beaac.png)

Attached two sets of diagnostics, after each behavior.

100% reproducible (3 of 3 tries)

  - [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 
![image](https://user-images.githubusercontent.com/14317301/73462012-1f002c00-4349-11ea-97dc-ac2dda0e9225.png) 
![image](https://user-images.githubusercontent.com/14317301/73462035-24f60d00-4349-11ea-86bd-ad65b6483fce.png)

### Information

  - Windows Version: Win 10 Enterprise 1909 18363.592
  - Docker Desktop Version: 2.2.0.0 (42247)
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: no

**Comment by simonferquel** [comment]
We have several fixes incoming. Would you have a repro case that we can run to confirm this is fixed ?

**Comment by brucelowekamp** [comment]
> We have several fixes incoming. Would you have a repro case that we can run to confirm this is fixed ?
I'm using the ola build tree I have in my Windows "linuxsrc" directory, but I assume any medium sized source tree will trigger the behavior.  Could repro it just with "tar" in a vanilla debian:latest image:

C:\Users\bloweka>docker run -t -i -v c:/Users/bloweka/linuxsrc:/home/lowekamp/src debian:latest
root@edb1ea987a7e:/home/lowekamp/src# tar czf - ola-bad > badtar.tar.gz
tar: ola-bad/.codespellignorelines: file changed as we read it
tar: ola-bad/.codespellignorewords: file changed as we read it
tar: ola-bad/.gitignore: file changed as we read it
...

root@edb1ea987a7e:/home/lowekamp/src/foo# cat ../badtar.tar.gz | tar xzf -
tar: ola-bad/.codespellignorelines: Cannot utime: Software caused connection abort
tar: ola-bad/.codespellignorelines: Cannot change ownership to uid 0, gid 0: Transport endpoint is not connected
tar: ola-bad/.codespellignorelines: Cannot change mode to rwxr-xr-x: Transport endpoint is not connected
tar: ola-bad/.codespellignorewords: Cannot open: Transport endpoint is not connected

tar: ola-bad: Cannot change mode to rwxrwxrwx: Transport endpoint is not connected
tar: Error is not recoverable: exiting now
root@edb1ea987a7e:/home/lowekamp/src/foo# ls
ls: cannot open directory '.': Transport endpoint is not connected

**Comment by simonferquel** [comment]
Can you point me to a place where I can get this build tree ?

**Comment by brucelowekamp** [comment]
It's https://github.com/OpenLightingProject/ola though as tar started throwing that error on the first file in it and the transport was disconnected on the untar by the third file, I have my doubts the particular tree is causing the problem here.  I'm not aware of anything odd about my setup, but as it's a corporate (AAD-joined) laptop subsribed to MS insider (internal) releases it might have a newer version of the OS, etc.

Bruce

**Comment by simonferquel** [comment]
Ok, I confirm I reproduce it, and it is fixed in our latest internal build

**Comment by kapsner** [comment]
Same issue here. When trying to extract tar-files from an R session running in a debian-based container on a Windows 10 Pro host gives me: 

```
...Cannot utime: Software caused connection abort
...Transport endpoint is not connected
```

**Comment by stephen-turner** [comment]
We have just released 2.2.0.3, which fixes this issue, so closing this ticket. In case any Edge users are watching this ticket, it will be on the Edge channel tomorrow too.
